docs/getting-started: Add deps for different linux distributions (#108)
unikraft/docs: content/en/docs/getting-started/_index.md

## Installing the CLI companion tool `kraft`

> The kraft tool and Unikraft build tool have a number of package requirements.

Reviewer: ```suggestion
In order to build and run Unikraft and to us its companion tool [`kraft`](https://github.com/unikraft/kraft), several tools and libraries being must be part of your system.
```

> The kraft tool and Unikraft build tool have a number of package requirements.

Reviewer: ```suggestion
In order to build and run Unikraft and to us its companion tool [`kraft`](https://github.com/unikraft/kraft), several tools and libraries being must be part of your system.
```

> The kraft tool and Unikraft build tool have a number of package requirements.

Reviewer: ```suggestion
In order to build and run Unikraft and to us its companion tool [`kraft`](https://github.com/unikraft/kraft), several tools and libraries being must be part of your system.
```

Follow the steps below according to your Linux distribution.

> ### Ubuntu, Debian and other APT-based distributions

Reviewer: ```suggestion
### Ubuntu, Debian and Other APT-based Distributions
```

> Please run the following command to install the requirements:

Reviewer: ```suggestion
Run the following command to install the requirements:
```

> ```shell-script

Reviewer: Use `console`: https://stackoverflow.com/a/49004070/4804196

  unzip \
  uuid-runtime \
  python3 \
  python3-setuptools \
  python3-pip
```

### Arch Linux, Manjaro and other Pacman-based distributions

Please run the following command to install the requirements:

```shell-script
pacman -Sy base-devel \
  ncurses \
  libyaml \
  flex \
  git \
  wget \
  socat \
  bison \
  unzip \
  python \
  python-setuptools \

>   python-pip

Reviewer 2: ```suggestion
  python-pip \
  qemu-system-x86
```

```

### Fedora and other dnf-based distributions

Please run the following command to install the requirements:

```shell-script
dnf install -y \
  @development-tools \
  ncurses-devel \
  libyaml \
  flex \
  git \
  wget \
  socat \
  bison \
  unzip \
  libuuid \
  python3 \
  python3-setuptools \

>   python3-pip

Reviewer 2: ```suggestion
  python3-pip \
  qemu-kvm \
  qemu-system-x86 \
  sgabios
```

  unzip \
  uuid-runtime \
  python3 \
  python3-setuptools \
  python3-pip \
  qemu-kvm \
  qemu-system-x86 \
  sgabios
```

### Arch Linux, Manjaro and other Pacman-based distributions

Please run the following command to install the requirements:

```shell-script
pacman -Sy base-devel \
  ncurses \
  libyaml \
  flex \
  git \
  wget \
  socat \
  bison \
  unzip \
  python \
  python-setuptools \
  python-pip \
  qemu-system-x86
```

### Fedora and other dnf-based distributions

Please run the following command to install the requirements:

```shell-script
dnf install -y \
  @development-tools \
  ncurses-devel \
  libyaml \
  flex \
  git \
  wget \
  socat \
  bison \
  unzip \
  libuuid \
  python3 \
  python3-setuptools \
  python3-pip \
  qemu-kvm \
  qemu-system-x86 \
  sgabios
```

> **Note: Since on Fedora the default `yacc` is [`Berkley Yacc`](https://en.wikipedia.org/wiki/Berkeley_Yacc), not [`GNU Bison`](https://www.gnu.org/software/bison/), you will need to manually set the `YACC` env variable `export YACC=/usr/bin/bison`.**

Reviewer 2: ```suggestion
**Note: Since on Fedora the default `yacc` is [`Berkeley Yacc`](https://en.wikipedia.org/wiki/Berkeley_Yacc), not [`GNU Bison`](https://www.gnu.org/software/bison/), you will need to manually set the `YACC` env variable `export YACC=/usr/bin/bison`.**
```

  unzip \
  uuid-runtime \
  python3 \
  python3-setuptools \
  python3-pip \
  qemu-kvm \
  qemu-system-x86 \
  sgabios
```

> ### Arch Linux, Manjaro and other Pacman-based distributions

Reviewer: ```suggestion
### Arch Linux, Manjaro and Other Pacman-based Distributions
```

> Please run the following command to install the requirements:

Reviewer: ```suggestion
Run the following command to install the requirements:
```

```console
$ pacman -Sy base-devel \
  ncurses \
  libyaml \
  flex \
  git \
  wget \
  socat \
  bison \
  unzip \
  python \
  python-setuptools \
  python-pip \
  qemu-system-x86
```

> ### Fedora and other dnf-based distributions

Reviewer: ```suggestion
### Fedora and other dnf-based Distributions
```

> Please run the following command to install the requirements:

Reviewer: ```suggestion
Run the following command to install the requirements:
```

```console
$ dnf install -y \
  @development-tools \
  ncurses-devel \
  libyaml \
  flex \
  git \
  wget \
  socat \
  bison \
  unzip \
  libuuid \
  python3 \
  python3-setuptools \
  python3-pip \
  qemu-kvm \
  qemu-system-x86 \
  sgabios
```

**Note: Since on Fedora the default `yacc` is [`Berkeley Yacc`](https://en.wikipedia.org/wiki/Berkeley_Yacc), not [`GNU Bison`](https://www.gnu.org/software/bison/), you will need to manually set the `YACC` env variable `export YACC=/usr/bin/bison`.**

> ### Install kraft using `pip3`

Reviewer: ```suggestion
### Install `kraft` Using `pip3`
```

> To then install kraft simply run:

Reviewer: ```suggestion
To then install `kraft` run:
```

> ```bash

Reviewer: Use `console`: https://stackoverflow.com/a/49004070/4804196

$ pip3 install git+https://github.com/unikraft/kraft.git
```

> You can then type kraft to see its help menu:

Reviewer: ```suggestion
You can then type `kraft` to see its help menu:
```

The simplest way to get the sources for, build and run an application is by running the following commands:

```console
$ kraft list
$ kraft up -p PLATFORM -m ARCHITECTURE APP
```

You can also build and run an applications from the existing templates by using the `-t` option.
For example, building and running the [helloworld app](https://github.com/unikraft/app-helloworld):

```console
$ kraft up -p kvm -m x86_64 -t helloworld helloworld
```

For more information about all the steps behind `kraft up`, check [Creating an App](docs/usage/init).

> For more information about the `up` command type kraft up -h.

Reviewer: ```suggestion
For more information about the `up` command type `kraft up -h`.
```

> For more information about kraft type kraft -h or read the documentation at Unikraft’s website.

Reviewer: ```suggestion
For more information about `kraft` type `kraft -h` or read the documentation at Unikraft's website.
```
Add link to the documentation.

The simplest way to get the sources for, build and run an application is by running the following commands:

> ```bash

Reviewer: Use `console`: https://stackoverflow.com/a/49004070/4804196

$ kraft list
$ kraft up -p PLATFORM -m ARCHITECTURE APP
```

You can also build and run an applications from the existing templates by using the `-t` option.
For example, building and running the [helloworld app](https://github.com/unikraft/app-helloworld):

```bash
$ kraft up -p kvm -m x86_64 -t helloworld helloworld
```

For more information about all the steps behind `kraft up`, check [Creating an App](docs/usage/init).
For more information about the `up` command type kraft up -h.
For more information about kraft type kraft -h or read the documentation at Unikraft’s website.
If you find any problems please fill out an issue.
Thank you!

## Building Unikraft via `Makefile`

To manually set up the build environment for the [helloworld app](https://github.com/unikraft/app-helloworld), we first have to create a specific directory structure:

```text
my-unikernel
	|-- apps
	|   `-- app-helloworld
	|-- libs
	`-- unikraft
```

We can do this by executing the following sequence of commands:

> ```bash

Reviewer: Use `console`: https://stackoverflow.com/a/49004070/4804196

For more information about all the steps behind `kraft up`, check [Creating an App](docs/usage/init).
For more information about the `up` command type kraft up -h.
For more information about kraft type kraft -h or read the documentation at Unikraft’s website.
If you find any problems please fill out an issue.
Thank you!

## Building Unikraft via `Makefile`

To manually set up the build environment for the [helloworld app](https://github.com/unikraft/app-helloworld), we first have to create a specific directory structure:

```text

>  my-unikernel

Reviewer: I would use the ASCII output of `tree` (`tree -L 1 --charset=ascii`).

      ├────apps
      │      └─app-helloworld     <- helloworld application
      ├─── libs                   <- additional libraries go here
      └─── unikraft               <- Unikraft core
```

We can do this by executing the following sequence of commands:

```bash
mkdir my-unikernel
cd my-unikernel
git clone https://github.com/unikraft/unikraft
mkdir libs
mkdir apps
cd apps
git clone https://github.com/unikraft/app-helloworld
cd app-helloworld
```

> All following configuration and build operations are done from within the helloworld application folder.

Reviewer: ```suggestion
All following configuration and build operations are done from within the `helloworld` application folder.
```

$(MAKECMDGOALS):
	@$(MAKE) -C $(UK_ROOT) A=$(PWD) L=$(LIBS) $(MAKECMDGOALS)
```

In the first two lines, we provide the paths to the Unikraft core repository and the directory where we can put additional libraries.
Since the helloworld application uses `nolibc` as C library, which comes with the Unikraft core, we do not have any external dependencies.
We thus do not have to download or reference any libraries in the `Makefile` and can leave the `LIBS` variable empty.

With the `Makefile` present and the directory structure correctly set up, we are able to configure the unikernel as the next step.
This will allow us to select the target architecture (e.g., aarch64 or x86-64) and platform (e.g., KVM or Xen).

> ```bash

Reviewer: Use `console`: https://stackoverflow.com/a/49004070/4804196

To add a library you simply clone it into the `libs` directory created earlier.
In this example, we add the lightweight IP (lwip) network stack [lib-lwip](https://github.com/unikraft/lib-lwip):

> ```bash

Reviewer: Use `console`: https://stackoverflow.com/a/49004070/4804196

$ cd my-unikernel
$ cd libs
$ git clone https://github.com/unikraft/lib-lwip
```

Next, we have reference the library in the `Makefile` of `app-helloworld` so that it becomes available for selection in the configuration:

```Makefile
UK_LIBS ?= $(PWD)/../../libs

LIBS := $(UK_LIBS)/lib-lwip
#LIBS := $(LIBS-y):$(UK_LIBS)/lib-X
```

We can now re-run

> ```bash

Reviewer: Use `console`: https://stackoverflow.com/a/49004070/4804196

$ make menuconfig
```

and select `lwip` in the `Library Configuration` sub-menu.
We can then rebuild the unikernel, by running:

> ```bash

Reviewer: Use `console`: https://stackoverflow.com/a/49004070/4804196
